SortApp/SortApp.BL/Sortings/Sorter.cs

```diff
@@ -9,6 +11,53 @@ namespace SortApp.BL.Sortings
 	public abstract class Sorter<T> where T : IComparable<T>
 	{
 		/// <summary>
+		/// Adds current iteration to iterations.
+		/// </summary>
+		/// <owner>Oleh Petrenko</owner>
+		/// <param name="arr">
+		/// The current state of the array.
+		/// </param>
+		/// <param name="firstIndex">
+		/// Index of first element which was moved.
+		/// </param>
+		/// <param name="secondIndex">
+		/// Index of second element which was moved.
+		/// </param>
+		protected void AddIteration(T[] arr, int firstIndex, int secondIndex)
+		{
+			Iteration curretnIteration = new Iteration
+			{
+				Number = this.Iterations.Count + 1,
+				IndexOfFirstElement = firstIndex,
+				IndexOfSecondElement = secondIndex
+			};
+
+			StringBuilder result = new StringBuilder(arr.Length);
+
+			foreach (T item in arr)
+			{
+				result.Append($"{item} ");
+			}
+
+			curretnIteration.Array = result.ToString();
+
+			this.Iterations.Add(curretnIteration);
+		}
+
+		/// <summary>
+		/// Gets or sets all sorting iterations.
+		/// </summary>
+		/// <owner>Oleh Petrenko</owner>
+		/// <param>
+		/// Sorting iterations.
+		/// </param>
+		public List<Iteration> Iterations
+		{
+			get;
+			protected set;
```

Comment:
Are there any sets of this array from child classes? If not - setter could be made private.